core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -437,6 +487,47 @@ private ExprValue exprAddDateDays(ExprValue date, ExprValue days) {
         : exprValue);
   }
 
+  /**
+   * CONVERT_TZ function implementation for ExprValue.
+   * Returns null for time zones outside of +13:00 and -12:00.
+   *
+   * @param startingDateTime ExprValue of DateTime that is being converted from
+   * @param fromTz           ExprValue of time zone, representing the time to convert from.
+   * @param toTz             ExprValue of time zone, representing the time to convert to.
+   * @return DateTime that has been converted to the to_tz timezone.
+   */
+  private ExprValue exprConvertTZ(ExprValue startingDateTime, ExprValue fromTz, ExprValue toTz) {
+    if (startingDateTime.type() == ExprCoreType.STRING) {
+      //try {
+      startingDateTime = exprDateTimeNoTimezone(startingDateTime);
+      //} catch (DateTimeParseException e) {
+      //  return ExprNullValue.of();
+      //}
+    }
+    try {
+      ZoneId convertedFromTz = ZoneId.of(fromTz.stringValue());
+      ZoneId convertedToTz = ZoneId.of(toTz.stringValue());
+
+      if (!isTimeZoneValid(convertedFromTz)
+          || !isTimeZoneValid(convertedToTz)) {
+        return ExprNullValue.of();
+      }
+
+      try {
+        ZonedDateTime zonedDateTime =
+            startingDateTime.datetimeValue().atZone(convertedFromTz);
+        return new ExprDatetimeValue(
+            zonedDateTime.withZoneSameInstant(convertedToTz).toLocalDateTime());
+      } catch (ExpressionEvaluationException e) {
+        return ExprNullValue.of();
+      }
```

Comment:
No need try-in-try, you still return NULL on exception.

Reply:
Fixed, added another exception to catch on try catch and removed internal.